src/Command/Ingester/CreateUserCommand.php

```diff
@@ -0,0 +1,461 @@
+            if ($lineItems->isEmpty()) {
+                throw new LineItemNotFoundException("No line items were found in database.");
+            }
+
+            $userIncrNo = $this->checkLineItemSlugUserExist($lineItems, $this->lineItemSlugs);
+
+            $userGroupAssignCount = 0;
+            if ($input->getOption(self::OPTION_GROUP_PREFIX)) {
+                $userGroupIds = $this->getLoadBalanceGroupID($input->getOption(self::OPTION_GROUP_PREFIX));
+                $userGroupAssignCount = (int)(count($this->userPrefix) * count($this->lineItemSlugs) * $this->batchSize / count($userGroupIds));
+            }
+
+            $automateCsvPath = $_ENV['AUTOMATE_USER_LIST_PATH'].date("Y-m-d");
+            $userCsvHead = ['username','password','groupId'];
+            $assignmentCsvHead = ['username','lineItemSlug'];
+            $noOfUsersCreated = 0;
+
+            $userAggregratedCsvDt = $assgAggregratedCsvDt = [];
+            $userDtoCollection = new UserDtoCollection();
+            $assignmentDtoCollection = new AssignmentDtoCollection();
+
+            if (!$this->filesystem->exists($automateCsvPath)) {
+                $this->filesystem->mkdir($automateCsvPath);
+            }
+            
+            foreach ($this->userPrefix as $prefix) {
+                $csvPath = $automateCsvPath. "/" . $prefix;
+                if (!$this->filesystem->exists($csvPath)) {
+                    $this->filesystem->mkdir($csvPath);
+                }
+                foreach ($this->lineItemSlugs as $lineSlugs) {
+                    $csv_filename = $lineSlugs."-".$prefix.".csv";
+                    $csvDt = $assignmentCsvDt = [];
+                    foreach (range(1, $this->batchSize) as $i) {
+                        $username = $lineSlugs."_".$prefix."_".((int)$userIncrNo[$lineSlugs] + (int)$i);
+                        $userPassword = $this->createUserPassword();
+                        $userGroupId = $input->getOption(self::OPTION_GROUP_PREFIX) ? 
+                            $this->createUserGroupId($userGroupIds, $userGroupAssignCount) : 
+                            '';
+                        
+                        $csvDt[] = [$username, $userPassword,$userGroupId];
+                        $assignmentCsvDt[] = [$username, $lineSlugs];
+                        
+                        $userDtoCollection->add($this->createUserDto($username, $userPassword, $userGroupId));
+                        $assignmentDtoCollection->add($this->createAssignmentDto($lineItems, $lineSlugs, $username));
+                        $noOfUsersCreated++;
+                    }
+                    $this->writeCsvData($path = $csvPath."/".$csv_filename, $userCsvHead, $csvDt);
+                    $this->writeCsvData($path = $csvPath."/"."Assignments-".$lineSlugs."-".$prefix.".csv", $assignmentCsvHead, $assignmentCsvDt);
+                    $userAggregratedCsvDt = array_merge($userAggregratedCsvDt,$csvDt);
```

Comment:
it is an overhead using array_merge in the loop.

Reply:
Removed the array_merge option and write or append data to csv files created